### PR TITLE
[VIM-3731] Add support for "jump to previous/next lowercase mark".

### PR DESCRIPTION
**Pull request by sdoerner**
Fixes [VIM-3731](https://youtrack.jetbrains.com/issue/VIM-3731/Add-support-for-jump-to-previous-next-lowercase-mark).

Let me know if I should do anything else with `Mark.PositionSorter`.  The two data structures are in different packages, so I wasn't sure if sharing the code would be desired or where you'd want the shared code to live.

### CODE REVIEW

**Comment by mbouron** [comment]
For what it's worth, I've been running this PR applied on top of latest master for the last few days and it works as expected.

**Comment by lippfi** [comment]
> For what it's worth, I've been running this PR applied on top of latest master for the last few days and it works as expected.

I've just checked out the PR to recheck my comment to be sure
Nope, it doesn't
Set multiple marks at the same place and then press ]`
Expected result: caret is moved to the next mark
Actual result: caret is stuck no matter how many times you press ]`

**Review by lippfi — CHANGES_REQUESTED** [review]

File: vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimMarkServiceBase.kt [hunk 1]
```diff
@@ -121,6 +121,36 @@ abstract class VimMarkServiceBase : VimMarkService {
...
+      caret.markStorage.getMarks().values.toSet()
+    }
+    val lowerCaseMarks = marks.filter { LOWERCASE_MARKS.contains(it.key) }
+      .sortedWith(Mark.PositionSorter)
+    // Use a fake mark to easily find the position of the caret in the list of marks.
+    val caretMark = createMark(caret, '[', caret.offset) ?: return null
+    val result = lowerCaseMarks.binarySearch(caretMark, Mark.PositionSorter)
+    val targetIndex = if (result >= 0) {
+      // Caret is on a mark.
+      result + count
```
[the remark this answers is not in the document]
Comment (sdoerner) [reply vim-engine/src/main/kotlin/com/maddyhome/idea/vim/api/VimMarkServiceBase.kt:139]: Thank you for pointing out this edge case. I fixed this in [ff6b4a1](https://github.com/JetBrains/ideavim/commit/ff6b4a1de2a8461b12d003c97d1c4d7f1bf84a6a) by removing marks with duplicate positions from the list initially. I also added two test cases to test this scenario.

Please take another look.

**Comment by sdoerner** [comment]
Pushed an update to the PR with a fix.

**Comment by lippfi** [comment]
Thank you for your work!
And thank you one more time for the tests, the number of tests is impressive.
